src/servermap.cpp

```diff
@@ -654,31 +687,16 @@ void ServerMap::loadBlock(std::string *blob, v3s16 p3d, MapSector *sector, bool
 		}
 
 		{
-			ScopeProfiler sp(g_profiler, "ServerMap: deSer block", SPT_AVG, PRECISION_MICRO);
-			block->deSerialize(is, version, true);
+			std::istringstream iss(blob, std::ios_base::binary);
+			deSerializeBlock(block, iss);
 		}
```

Comment:
Don't you want a try catch here?
This can still throw, leaving `block` null.

Reply:
it is in a try-catch?

Reply:
I mean a smaller try catch block.

(Ah, but I'm seeing now, block gets `createBlankBlockNoInsert`-ed before an exception can happen, so it won't be null. But making the try catch block as small as possible would still be cleaner.)